File: csharp-snippets/Legacy/Employee.cs
namespace Richargh.BillionDollar.Legacy
{
    public class Employee
    {

        public EmployeeId Id { get;  }
        public string Name { get; }
        public Notebook Notebook { get; set; }
        
        public Employee(EmployeeId id, string name, Notebook notebook)
        {
            Id = id;
            Name = name;
            Notebook = notebook;
        }
        
    }
}

File: java-rent-nullaway/src/main/java/de/richargh/billiondollar/people/exposed/Employee.java
package de.richargh.billiondollar.people.exposed;

import org.jspecify.annotations.Nullable;

public record Employee(
        EmployeeId id,
        String name,
        @Nullable Notebook notebook,
        @Nullable Budget budget) {

    public Employee withNotebook(Notebook notebook, Budget budget) {
        return new Employee(id, name, notebook, budget);
    }

    public Employee withoutNotebook() {
        return new Employee(id, name, null, budget);
    }

    public boolean hasNotebook() {
        return notebook != null;
    }
}
